zebra-chain/src/sapling/keys.rs

```diff
@@ -837,14 +837,20 @@ impl fmt::Debug for TransmissionKey {
 
 impl Eq for TransmissionKey {}
 
-impl From<[u8; 32]> for TransmissionKey {
-    /// Attempts to interpret a byte representation of an
-    /// affine point, failing if the element is not on
-    /// the curve or non-canonical.
+impl TryFrom<[u8; 32]> for TransmissionKey {
+    type Error = &'static str;
+
+    /// Attempts to interpret a byte representation of an affine Jubjub point, failing if the
+    /// element is not on the curve, non-prime, the identity, or non-canonical.
     ///
     /// https://github.com/zkcrypto/jubjub/blob/master/src/lib.rs#L411
-    fn from(bytes: [u8; 32]) -> Self {
-        Self(jubjub::AffinePoint::from_bytes(bytes).unwrap())
+    fn try_from(bytes: [u8; 32]) -> Result<Self, Self::Error> {
+        let affine_point = jubjub::AffinePoint::from_bytes(bytes).unwrap();
+        if affine_point.is_prime_order().into() {
```

Comment:
(I realized this is a draft, but I decided to take a look out of curiosity :laughing: )

I'm not sure if this is what the spec wants, since the spec allows composite order points which are not allowed by this code (i.e. "not small order" and "prime order" are different)

We also already do the consensus rule validation, though I agree it would be better to move them here:

https://github.com/ZcashFoundation/zebra/blob/2f46d698dd86e40e7928bddca948f5be14667934/zebra-consensus/src/transaction/check.rs#L135

Reply:
After studing this a little bit, you were completely right. In this particular case we really do want to check that it's prime order due to https://zips.z.cash/zip-0216. It's other stuff like cv and rk that we can just check for small order (which we already do in that code I linked, but we can move the check to parsing)

I think we could postpone this particular check (TransmissionKey) since it seems it's only relevant for wallets. But if it's simple it's probably better to do it now.